enterprise/frontend/src/embedding-sdk/store/auth/auth.ts

```diff
@@ -101,37 +105,17 @@ export const refreshTokenAsync = createAsyncThunk(
     url: MetabaseAuthConfig["metabaseInstanceUrl"],
     { getState },
   ): Promise<MetabaseEmbeddingSessionToken | null> => {
-    const customGetRefreshToken =
-      getFetchRefreshTokenFn(getState() as SdkStoreState) ?? null;
+    const state = getState() as SdkStoreState;
+
+    if (getIsNewIframeEmbeddingAuth(state)) {
+      return requestSessionTokenFromEmbedJs();
+    }
```

Comment:
If we are in the new iframe embedding, we want to do authentication outside of the iframe. This asks the embed.js in the parent window for the session token.